pkg/api/options.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	"time"
+)
+
+// Option is an additional argument to method New to change the behavior
+// of API processing.
+type Option interface {
```

Comment:
Nit: this could probably be simply a function, e.g. `type Option func(*Config)` (but this is fine too, especially if you foresee to add more methods)

Reply:
I do not like dynamically created functions passed through packages. For example, I cannot compare functions, serialize them or at least to print its value. I understand all these is not used. I don't mind to change it to function-based approach if you still want to. But until then I would like to preserve current approach :)